client/tests/webdriver/pages/searchFilters/taskFilter.page.js

```diff
@@ -0,0 +1,84 @@
+import Page from "./../page"
+
+class TaskFilter extends Page {
+  async openTaskFilter() {
+    const searchLink = await browser.$(
+      ".search-popover-target.bp5-popover-target"
+    )
+    await searchLink.waitForDisplayed()
+    await searchLink.click()
+
+    const reportsButton = await browser.$('.btn-group > button[value="REPORTS"')
+    await reportsButton.waitForDisplayed()
+    await reportsButton.click()
+
+    const removeFilterButtons = await browser.$$(
+      '.form-group button[title="Remove this filter"'
+    )
+    removeFilterButtons.forEach(async removeFilterButton => {
```

Comment:
This forEach loop fails to wait for the promises inside the loop to resolve. You could rewrite it as:
```javascript
    for (const removeFilterButton of removeFilterButtons) {
      await removeFilterButton.waitForDisplayed()
      await removeFilterButton.click()
    }
```